src/SAWScript/Proof.hs

```diff
@@ -176,6 +177,21 @@ simplifyProp sc ss (Prop tm) =
   do tm' <- rewriteSharedTerm sc ss tm
      return (Prop tm')
 
+hoistIfsInGoal :: SharedContext -> Prop -> IO Prop
+hoistIfsInGoal sc (Prop p) = do
+  let (args, body) = asPiList p
+  body' <-
+    case asEqTrue body of
+      Just t -> pure t
+      Nothing -> fail "hoistIfsInGoal: expected EqTrue"
+  ecs <- traverse (\(nm, ty) -> scFreshEC sc nm ty) args
```

Comment:
We should really check to make sure that there aren't dependencies among these types, otherwise we will have dangling variables.  Either that, or we should handle it properly by substituting into the types of later variables as well.

Reply:
Looks like we use this same pattern elsewhere.  Let's not hold up this PR for this, but it's definitely something that should get fixed.

Reply:
Looks like doing this properly is over my head for now. I presume this is not critical at the moment since you approved the changes. Is that right?

Reply:
Yeah, I don't think we need to fix this right now. CF https://github.com/GaloisInc/saw-script/issues/1316